### ISSUE TITLE
[Bug] The ESP creates an open AP with the name ESP-*

### ISSUE THREAD

**Issue by pcmester** [Neutrino-1/Hack-Casio-Calculator#1]
Hi there,
I fixed all of the errors, but the problem is, that the code does not want to function well. Something is wrong, but in the debugging process, I could not figure out, what causes this issue. The ESP instead of connecting to the provided WiFi network, creates an AP with no webserver on it. Could somebody help figuring out what's the issue?

Yours sincelery, 
pcmester

**Comment by Neutrino-1** [comment]
can you send the error report?
Hotspot turns on only during GPIO pin 12 is pulled LOW, during boot.
So, make sure you use a pull resistor on GPIO pin 12,

**Comment by pcmester** [comment]
If I turn on serial logging (and remove the uncomment from every serial.print command), the messages are: 
Started!
The exact case is, that I don't have the magnetic sensors in my hand, so I would like to do an experiment without it. Can it be the problem?

**Comment by Neutrino-1** [comment]
Yes, that can be one of the problem, use a 10k resistor between the pin GPIO 12 and +3.3V and try again.

**Comment by pcmester** [comment]
Thanks! I will try it

**Comment by pcmester** [comment]
Sadly still nothing, the device just creates an unencrypted network with the name ESP-*****
If I move WiFi connection to setup, it connects to WiFi, but that's it. The code doesn't continues. What could be the problem?

**Comment by Neutrino-1** [comment]
Could you send the corrected code in a temporary link? and which module you are using 12E or nodeMCU?

**Comment by pcmester** [comment]
Of course I can, here you go: https://uploadfiles.io/w9lu34b4
Currently, I'm doing my experiment with the NodeMCU. This is the 4 error-corrected, in other respects unmodified code.

**Comment by Neutrino-1** [comment]
![git](https://user-images.githubusercontent.com/15338998/82166474-26d2e480-98d6-11ea-8d69-eb3558f39499.png)
The code you sent works fine, even though _handleSettingUpdate()_ fix isn't right.

If your using NodeMCU, GPIO pin 12 corresponds to D6. So make sure that is pulled down LOW when you want to create AP.
Make sure the board is selected to **Generic Esp8266 module** rather than **NodeMCU** and the flash size **4mb (FS: 2mb, OTA: 1019KB)** is selected.

**Comment by pcmester** [comment]
I tried all of the fixes you said, and the case still the same: An AP called ESP-58B3** appears, and the message on the console is just started. I checked everything, and still couldn't find the problem

**Comment by pcmester** [comment]
Finally, I decided to continue my investigation, by buying the missing part, I mean, the hall effect sensor. Do you think that a linear hall effect sensor, exactly a Honeywell SS49E will suit the case?

**Comment by Neutrino-1** [comment]
I don't this that might help, because as far as I know "Honeywell SS49E" output's analog value at pin 3.
Go with "A3144 hall effect sensor" this output's a digital output.

**Comment by pcmester** [comment]
Thank you, I will check it!

**Comment by pcmester** [comment]
So I bought the hall effect sensors, and a new ESP8266 board (Wemos D1), and the code still producates the same issue. What other thing could cause an issue like this?

**Comment by bitsinside** [comment]
> 
> 
> Finally, I decided to continue my investigation, by buying the missing part, I mean, the hall effect sensor. Do you think that a linear hall effect sensor, exactly a Honeywell SS49E will suit the case?

buy reed switch
